lib/marked.js

```diff
@@ -762,8 +703,18 @@ InlineLexer.prototype.output = function(src) {
       src = src.substring(cap[0].length);
       this.inLink = true;
       href = cap[2];
-      href = href[0] === '<' ? href.substring(1, href.length - 1) : href;
-      title = cap[3] ? cap[3].substring(1, cap[3].length - 1) : cap[3];
+      if (this.options.pedantic) {
```

Comment:
We don't test for pedantic much, and I don't see it well documented anywhere. The only docs I could find are in the man page.

Can you explain when I should test for pedantic in the future?

Reply:
If I understand this code right, without `pedantic` we will set `href` but not `title`. If my understanding is correct, why is this behavior desirable?

Reply:
1.  The regexes are grouped based on the options. The link regex that you fixed was in the pedantic group

    ```js
    inline.pedantic = merge({}, inline.normal, {
    ```

2.  The title is set in the else clause. The regex for a non-pedantic link is `/^!?\[(label)\]\(href(?:\s+(title))?\s*\)/` which is extremely complex when constructed and probably also error prone.

Reply:
here is the actual non-pedantic link regex:

```js
/^!?\[((?:\[[^\[\]]*\]|\\[\[\]]?|`[^`]*`|[^\[\]\\])*?)\]\(\s*(<(?:\\[<>]?|[^\s<>\\])*>|(?:\\[()]?|\([^\s\x00-\x1f()\\]*\)|[^\s\x00-\x1f()\\])*?)(?:\s+("(?:\\"?|[^"\\])*"|'(?:\\'?|[^'\\])*'|\((?:\\\)?|[^)\\])*\)))?\s*\)/
```

Reply:
I don't see why this shouldn't be part of the 0.4.0 release

Reply:
Sounds good. Just wasn't sure where the pedantic situation was.

Reply:
> The title is set in the else clause

Sorry I misspoke earlier.

On the path `!pedantic` it gets set.

I am thinking of the path:

`pedantic && !link`

and it seems like we then have `href` defined but not `title`?

Reply:
The regex that you used before when there was no title (`/^(<?[\s\S]*>?)/`) literally matches any string and since `href` is already set to the whole string there is no need to change it.

Reply:
Further down we then set the `title` field in the returned object based on the (undefined?) value of title. Do we want to set a default value for `title`, e.g. `''`?

Reply:
My two cents would be an empty value of expected type. Avoid null and undefined checks and possibilities.